File: java23/Sublist.java
package java23;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Sublist {
    public static List<List<String>> sublist(List<String> list) {
        List<List<String>> newList = new ArrayList<>();
        List<String> sublist = new ArrayList<>();

        for (int i = 0; i < list.size(); i++) {
            if (i == 0 || !list.get(i).equals(list.get(i - 1)) ) {
                if (!sublist.isEmpty()) {
                    newList.add(sublist);
                }
                sublist = new ArrayList<>();
            }
            sublist.add(list.get(i));
        }

        if (!sublist.isEmpty()) {
            newList.add(sublist);
        }

        return newList;
    }

    public static void main(String[] args) {

        List<String> exampleList = new ArrayList<>(Arrays.asList("a", "a", "a", "b", "b", "c", "d", "a", "a", "e"));
        List<List<String>> finalList = sublist(exampleList);
        System.out.println(finalList);
    }
    
}

File: python23/sublist.py
#Problem 9

def new_list(one):

    sublist = []
    new_list = []

    for i in range(len(one)):

        if i == 0 or (one[i] != one[i-1]):
            if sublist:
                new_list.append(sublist)
                sublist = []

        sublist.append(one[i])
    
    if sublist:
        new_list.append(sublist)

    return new_list

example_list = ['a', 'a', 'a', 'b', 'b', 'c', 'b', 'b', 'd']
final_list = new_list(example_list)
print(final_list)
